ext/pgsql/pgsql.c

```diff
@@ -2455,39 +2524,40 @@ PHP_FUNCTION(pg_lo_open)
 		RETURN_THROWS();
 	}
 
-	pgsql_lofp = (pgLofp *) emalloc(sizeof(pgLofp));
+	object_init_ex(return_value, pgsql_lob_ce);
+	pgsql_lofp = Z_PGSQL_LOB_P(return_value);
```

Comment:
I think it would be cleaner to move this to the bottom of the function, so you don't need to dtor on failure.

Reply:
I moved it to the bottom, and I removed the `return`s from the positive branches.